### DISCUSSION TITLE
[EMSUSD-1046] Transform manipulator doesn't respect pivot position

### DISCUSSION THREAD

**Discussion by Ark-Dem** [Autodesk/maya-usd#3618] [General]
(Maya 2024.2, Win10, mayaUSD 0.27.0)

For maya object a transform manipulator follows its pivot, but after exporting to usd it appears at the origin while transformations still work correctly around this custom pivot.

The only option right now seems to be to move the asset so that custom pivot is at the origin (freeze transformations have to be done which can be a big pain) before exporting... or to suck up.

It would be great to make transform manipulator behave the same way for usd objects as it does for maya objects.

Here's a visual example of the issue...

That's maya "door" object with pivot set to "hinges":
![maya](https://github.com/Autodesk/maya-usd/assets/154688367/64576d8a-5ac4-4fc4-ba20-3fd048eac9a6)

After exporting to USD it works like this:
![usd](https://github.com/Autodesk/maya-usd/assets/154688367/74c4a96b-491b-43b4-ac5f-fbc47557fbf5)

The only solution I've found is to move it to the origin and freeze transforms before exporting to usd (not ideal for many assets):
![image](https://github.com/Autodesk/maya-usd/assets/154688367/f885f065-a770-4e79-b44e-8cb2a1cd90dc)

**Comment by wallworm** [comment]
We will look into this. Thank you for reporting.

**Comment by santosd** [comment]
Hi @Ark-Dem I tested this out on my end and it seems to be working as expected for me. One thing that could be happening is that when you create the stage from file, it automatically selects the stage transform instead of the mesh prim.
In the attached video, I show export with default options and as you can see that works as expected. I also show an export without merging the transform so then when you create a stage from that file you notice that there is a transform prim above the mesh prim and that transform will contain the pivot in the correct place. Either way though it seems to be working on my end. Another thing to make note of is that my rotate, translate tool, in Maya is set to Object orientation on export. Is that the same for you?

https://github.com/Autodesk/maya-usd/assets/29235905/af471f44-2b60-4f1b-a07e-4f0821109213

**Comment by Ark-Dem** [comment]
Hi @santosd, that's because you didn't Freeze Transforms of your cube before exporting. This way it matches the "only solution" I've mentioned above - set the pivot where you need, move the object so that this pivot is in the origin, freeze transforms, then move the object back to its original place - this way Transform Manipulator looks to be in proper place, but it's technically in (0,0,0)... it's just the whole object with pivot is translated, it has non-zero transforms.

That doesn't match Maya behavior where you can have zero transforms and manipulator still follows the custom pivot position.

Try Freeze Transform right before exporting your cube to usd and you will get my situation.

**Reply by santosd** [reply]
I see, that makes sense. I will give this a try.

**Comment by Ark-Dem** [comment]
I've noticed this: 

if the object has rotatePivot and scalePivot with different values (that happens when you move pivot for the object with scale other than (1,1,1)), usd writes pivot in this manner:
```
float3 xformOp:translate:rotatePivot = (0, 0, -0.5)
float3 xformOp:translate:scalePivot = (0, 0, -1.5)
```
and everything works fine, manipulator matches.

If rotatePivot and scalePivot have the same value (moving pivot for non-scaled object or making Freeze Transforms), the pivot is written like this:
`float3 xformOp:translate:pivot = (0, 0, -0.5)`
and manipulator doesn't follow it.

So, it seems that manipulator respects xformOp:translate:rotatePivot & xformOp:translate:scalePivot, but doesn't respect xformOp:translate:pivot.

**Comment by neilrubie** [comment]
Just wanted to add that we are seeing the same behaviour as @Ark-Dem. It's been that way for many versions of the plugin now, and we've clumsily got around it by telling our artists not to modify any pivots. Or if they do, they must use Maya's Bake Pivot command on the object. This is quite limiting to our desired workflow, it would be great to get this addressed.

**Comment by santosd** [comment]
I was able to reproduce the issue and do see that the translate is being captured but it seems that the translate handles are not being displayed where the pivot should be. I will log this issue internally for the team to take a look at.

**Comment by santosd** [comment]
Logged EMSUSD-1046 internally for the team to take a look at and see if we can get this fixed.
